opensrp-reveal/src/main/java/org/smartregister/reveal/view/ListTasksActivity.java

```diff
@@ -308,16 +319,29 @@ public void setGeoJsonSource(@NonNull FeatureCollection featureCollection, Featu
             geoJsonSource.setGeoJson(featureCollection);
             if (operationalArea != null) {
                 CameraPosition cameraPosition = mMapboxMap.getCameraForGeometry(operationalArea.geometry());
+                if (getInterventionLabel() == R.string.focus_investigation) {
+                    Feature indexCase = getIndexCase(mMapboxMap.getStyle().getSourceAs(getContext().getString(R.string.reveal_datasource_name)));
+                    if (indexCase != null) {
+                        cameraPosition = mMapboxMap.getCameraForGeometry(indexCase.geometry());
+                    }
+                } else {
+                    removeCaseIndexBoundary(mMapboxMap.getStyle());
+                }
+
                 if (cameraPosition != null) {
                     mMapboxMap.setCameraPosition(cameraPosition);
                 }
+
                 if (boundaryLayer == null) {
                     boundaryLayer = createBoundaryLayer(operationalArea);
                     kujakuMapView.addLayer(boundaryLayer);
-
                 } else {
                     boundaryLayer.updateFeatures(FeatureCollection.fromFeature(operationalArea));
                 }
+
+                if (getInterventionLabel() == R.string.focus_investigation) {
+                    addCaseIndexBoundary(mMapboxMap, getContext());
```

Comment:
Maybe you should use the above example for the boundary layer. Create layer is not already created, otherwise just update the GeoJSON Source

Reply:
Don't think I understand what the issue is here, could you rephrase?

Reply:
Create the index case layers once, then update the source just like how the operational area boundaries and labels are drawnin the above snippet.

```
            if (boundaryLayer == null) {
                    boundaryLayer = createBoundaryLayer(operationalArea);
                    kujakuMapView.addLayer(boundaryLayer);

                } else {
                    boundaryLayer.updateFeatures(FeatureCollection.fromFeature(operationalArea));
                }
```